storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -512,10 +514,146 @@ func (s *peersSource) ReadIndex(
 	shardsTimeRanges result.ShardTimeRanges,
 	opts bootstrap.RunOptions,
 ) (result.IndexBootstrapResult, error) {
-	// FOLLOWUP(r): implement the peers source returning
-	// index segments that are streamed from peers as required.
-	// Try to cache some series metadata on the peers source itself
-	// from work done in ReadData(...) to help avoid rereading as
-	// much as possible.
-	return result.NewIndexBootstrapResult(), nil
+	// FOLLOWUP(r): Try to reuse any metadata fetched during the ReadData(...)
```

Comment:
Think maybe we should make a master issue for index-bootstrap performance improvements? I'm gonna have some for the commit log as well.

Reply:
Yeah, we can potentially pile it onto this one:
https://github.com/m3db/m3db/issues/585

Reply:
yeah lets throw it in there, I'll put my stuff there too

Reply:
Added it to there, and added the commit log bootstrapper too.